packages/e2e-tests/specs/experiments/customizing-widgets.test.js

```diff
@@ -0,0 +1,427 @@
+		const widgetsPanel = await find( {
+			role: 'heading',
+			name: /Widgets/,
+			level: 3,
+		} );
+		await widgetsPanel.click();
+
+		const footer1Section = await find( {
+			role: 'heading',
+			name: /Footer #1/,
+			level: 3,
+		} );
+		await footer1Section.click();
+
+		const documentTools = await find( {
+			role: 'toolbar',
+			name: 'Document tools',
+		} );
+
+		const addBlockButton = await find(
+			{
+				role: 'button',
+				name: 'Add block',
+			},
+			{ root: documentTools }
+		);
+		await addBlockButton.click();
+
+		const paragraphOption = await find( {
+			role: 'option',
+			name: 'Paragraph',
+		} );
+		await paragraphOption.click();
+
+		const addedParagraphBlock = await find( {
+			role: 'group',
+			name: /^Empty block/,
+		} );
+		await addedParagraphBlock.focus();
+
+		await page.keyboard.type( 'First Paragraph' );
+
+		await addBlockButton.click();
+
+		const headingOption = await find( {
+			role: 'option',
+			name: 'Heading',
+		} );
+		await headingOption.click();
+
+		const addedHeadingBlock = await find( {
+			role: 'group',
+			name: 'Block: Heading',
+			level: 2,
+		} );
+		await addedHeadingBlock.focus();
+
+		await page.keyboard.type( 'My Heading' );
+
+		const inlineAddBlockButton = await find( {
+			role: 'combobox',
+			name: 'Add block',
+			haspopup: 'menu',
+		} );
+		await inlineAddBlockButton.click();
+
+		const inlineInserterSearchBox = await find( {
+			role: 'searchbox',
+			name: 'Search for blocks and patterns',
+		} );
+
+		await expect( inlineInserterSearchBox ).toHaveFocus();
+
+		await page.keyboard.type( 'Search' );
+
+		const searchOption = await find( {
+			role: 'option',
+			name: 'Search',
+		} );
+		await searchOption.click();
+
+		const addedSearchBlock = await find( {
+			role: 'group',
+			name: 'Block: Search',
+		} );
+
+		const searchTitle = await find(
+			{
+				role: 'textbox',
+				name: 'Label text',
+			},
+			{ root: addedSearchBlock }
+		);
+		await searchTitle.focus();
+
+		await page.keyboard.type( 'My ' );
+
+		const findOptions = {
+			get root() {
+				return find( {
+					name: 'Site Preview',
+					selector: 'iframe',
+				} );
+			},
+		};
+
+		// Expect the paragraph to be found in the preview iframe.
+		await expect( {
+			text: 'First Paragraph',
+			selector: '.widget-content p',
+		} ).toBeFound( findOptions );
+
+		// Expect the heading to be found in the preview iframe.
+		await expect( {
+			role: 'heading',
+			name: 'My Heading',
+			selector: '.widget-content *',
+		} ).toBeFound( findOptions );
+
+		// Expect the search box to be found in the preview iframe.
+		await expect( {
+			role: 'searchbox',
+			name: 'My Search',
+			selector: '.widget-content *',
+		} ).toBeFound( findOptions );
+
+		expect( console ).toHaveWarned(
+			"The page delivered both an 'X-Frame-Options' header and a 'Content-Security-Policy' header with a 'frame-ancestors' directive. Although the 'X-Frame-Options' header alone would have blocked embedding, it has been ignored."
```

Comment:
When does this warning happen? I don't see it in the browser console when adding blocks manually.

Reply:
I think it's a puppeteer thing, we might have to enable some flag to disable the warning. Not sure if it's safe to do so though.